BusinessLayer/JsonForwarder/JsonForwarder.cpp

```diff
@@ -53,11 +53,16 @@ void JsonForwarder::startForwardingData()
     readTimer_->start();
 }
 
-void JsonForwarder::forwardData()
+void JsonForwarder::handleTimeout()
+{
+    QString currentTime = QDateTime::currentDateTime().toUTC().toString(MYSQL_DATE_FORMAT);
+    forwardData(currentTime);
+}
+
+void JsonForwarder::forwardData(QString& currentTime)
```

Comment:
Cool, I feel like you probably understand at least a little bit more what it means to write testable code now ;) 

I would actually suggest passing in QTime instead of passing in a string, that is the intent here anyways, to pass in a time. Maybe one day forwardData will need to do some manipulation of the time.